hyperSpec/vignettes/vignette-default-settings.R

```diff
@@ -68,12 +70,30 @@ set_trellis_layout_hw_custom <- function() {
 set_trellis_layout_hw_custom()
 
 # knitr Settings =============================================================
+
+# Notes on Figure Sizes.  There are (at least) 2 common figure aspect ratios in
+# the vignettes.  One is a full width figure, suitable for spectra or spectra-like
+# plots.  The other is a square plot, typically used by maps.
+# Figure size is controlled by the hook defined just below, and notice that
+# in the opts_chunk sq.fig = FALSE as the default.
+
+knitr::opts_hooks$set(sq.fig = function(options) {
```

Comment:
Brilliant idea to use hooks.

Reply:
Thanks.  As the commit history shows, it took me quite a while to get this just right.  In Yihui's books there are several near-examples that misled me for a while.  But this works pretty nicely, you only have to set in a single place, which I think was the objective.

Reply:
Does the setting this hook allow having custom-sized figures?

Reply:
It seems it doesn't. Compare:

1. http://cbeleites.github.io/hyperSpec/articles/chondro.html#fig:irregular (original)
2. http://cbeleites.github.io/hyperSpec/vig/articles/chondro.html#fig:irregular (with hook)

I'd prefer to have the possibility to use custom-sized plots.

Reply:
Well, I guess one could go back to setting default `fig.width` and `fig.height` and then changing them in the chunks where some other dimensions are desired.  From my experience getting this piece to work, you have to employ `options$fig.width` etc so it would probably be a lot of work to override but I'll think about it.  In the meantime I think we should move forward.